conjur/user/user_logic.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+
+"""
+UserLogic module
+
+This module is the business logic for handling all user-related activity
+"""
+
+# Builtins
+import logging
+import requests
+
+# Internals
+from conjur.credentials_from_file import CredentialsFromFile
+from conjur.errors import InvalidOperationException, OperationNotCompletedSuccessfullyException
+
+
+class UserLogic:
+    """
+    UserLogic
+
+    This class holds the business logic for handling user activity
+    """
+    def __init__(self, conjurrc_data, credentials_from_store, client, resource):
+        self.conjurrc_data=conjurrc_data
+        self.credentials_from_store=credentials_from_store
+        self.client=client
+        self.resource=resource
+
+    # pylint: disable=logging-fstring-interpolation,line-too-long
+    def rotate_api_key(self, resource_to_update):
+        """
+        Method to trigger two types of API key actions.
+        1. The user can either rotate their own API key or
+        2. Rotate another users API key
+        """
+        loaded_credentials = self.extract_credentials_from_credential_store()
+        user_from_credential_store = loaded_credentials['login_id']
+        if resource_to_update == user_from_credential_store:
+            raise InvalidOperationException("To rotate the API key of the currently logged-in user "\
+                                   "use this command without any flags or options")
+        if resource_to_update is not None:
```

Comment:
I would check for `is None` first, and leave the `else` to refer to `is not None`.
```
if a is None:
  ...
else:
  ...
```
is more readable than:
```
if a is not None:
  ...
else:
  ...
```
That is b/c `else` after `not None` is double negation, and you should avoid it as much as possible.

Reply:
Totally agree but wanted to handle the shorter logic first so it more readable. So it becomes a question of either short logic first or double negative. Do you still recommend going with `if a is None: ...long logic else: short logic`